metricbeat/module/system/socket/config.go

```diff
@@ -0,0 +1,28 @@
+package socket
+
+import "time"
+
+// Config is the configuration specific to the socket MetricSet.
+type Config struct {
+	ReverseLookup ReverseLookupConfig `config:"socket.reverse_lookup"`
+}
+
+// ReverseLookupConfig contains the configuration that controls the reverse
+// DNS lookup behavior.
+type ReverseLookupConfig struct {
+	Enabled    *bool         `config:"enabled"`
+	SuccessTTL time.Duration `config:"success_ttl"`
+	FailureTTL time.Duration `config:"failure_ttl"`
+}
+
+// IsEnabled returns true if enabled is not defined or is set to true.
+func (c ReverseLookupConfig) IsEnabled() bool {
+	return c.Enabled == nil || *c.Enabled == true
```

Comment:
Does this mean that by default reverse lookup is enabled, right? I'm not sure about enabling that by default since it can cause performance issues, especially since there's no configurable timeout on the lookup, right?